File: api/server/src/main/java/su/plo/voice/api/server/audio/source/ServerStaticSource.java
package su.plo.voice.api.server.audio.source;

import org.jetbrains.annotations.NotNull;
import su.plo.lib.api.server.world.ServerPos3d;
import su.plo.voice.proto.data.audio.source.StaticSourceInfo;

public interface ServerStaticSource extends ServerPositionalSource<StaticSourceInfo> {

    void setPosition(@NotNull ServerPos3d position);
}

File: api/server/src/main/kotlin/su/plo/voice/api/server/audio/source/ServerStaticSource.kt
package su.plo.voice.api.server.audio.source

import su.plo.slib.api.server.position.ServerPos3d
import su.plo.voice.proto.data.audio.source.StaticSourceInfo

/**
 * Represents a static audio source.
 */
interface ServerStaticSource : ServerPositionalSource<StaticSourceInfo> {

    /**
     * Gets or sets the position of this audio source.
     *
     * @return The position.
     */
    override var position: ServerPos3d
}
